src/activities/quizzes/attempts/QuizAttemptsEntity.js

```diff
@@ -83,43 +83,85 @@ export class QuizAttemptsEntity extends Entity {
 		return attemptsAllowed !== this.attemptsAllowed();
 	}
 
+	/**
+	 * Returns an update attempts action if one exists
+	 * @param {number} attemptsAllowed number of attempts allowed
+	 */
+
+	_generateAttemptsAllowedAction(attemptsAllowed) {
+		let action;
+		if (!this.canUpdateAttemptsAllowed()) {
+			return;
+		}
+
+		if (this._entity) {
+			action = this._entity.getActionByName(Actions.quizzes.attempts.updateAttemptsAllowed);
+		}
+
+		if (!action) {
+			return;
+		}
+
+		const fields = [
+			{ name: 'attemptsAllowed', value: attemptsAllowed },
+		];
+
+		return { action, fields };
+
+	}
+
 	async setAttemptsAllowed(attemptsAllowed) {
 		if (!attemptsAllowed || !this._hasAttemptsAllowedChanged(attemptsAllowed)) return;
 		if (!this.canUpdateAttemptsAllowed()) {
 			return;
 		}
-		const {action, fields } = this._generateAttemptsAllowedAction(attemptsAllowed);
+		const {action, fields} = this._generateAttemptsAllowedAction(attemptsAllowed);

```

Comment:
This needs to be 
`const {action, fields} = this._generateAttemptsAllowedAction(attemptsAllowed) || {};`
Otherwise it is a `TypeError` as it can't destructure action and fields.

Reply:
I've applied this to the `overallGradeCalculation` as well